### PR TITLE
Remove `--DRT-oncycle=deprecate` option

### PR DESCRIPTION
**Pull request by dkorpel**
Remove the old "deprecate" mode for cycle handling: https://github.com/dlang/druntime/pull/1720

Reduces the size of a `void main() {}` program. Results from `dmd test.d && du test`:

|              | before | after |
|--------------|--------|-------|
|BUILD=debug   |2604 Kb |2588 Kb|
|BUILD=release |888 Kb  |884 Kb |

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
Looks fine to me, cc @schveiguy

**Comment by dkorpel** [comment]
Looks like it was `test_cycles` after all, but the output continued after that so it looked like it had to do with code coverage or the gc benchmark on some CIs.

File: test/cycles/Makefile [hunk 1]
```diff
@@ -14,8 +14,8 @@ $(ROOT)/cycle_abort.done: RETCODE=1
 $(ROOT)/cycle_abort.done: LINES=7
 $(ROOT)/cycle_print.done: RETCODE=0
 $(ROOT)/cycle_print.done: LINES=6
-$(ROOT)/cycle_deprecate.done: RETCODE=0
-$(ROOT)/cycle_deprecate.done: LINES=4
+$(ROOT)/cycle_deprecate.done: RETCODE=1
```
Comment (ljmf00) [inline test/cycles/Makefile:17]: Shouldn't this be deleted? It doesn't seem to reflect the reality

Comment (dkorpel) [reply test/cycles/Makefile:17]: It does, it's the same as abort now + 1 line for the message that the option is now removed.

**Comment by ljmf00** [comment]
> > Isn't this worth deprecating instead of just removing?
> 
> It was a helper for transitioning code that relied on the old faulty cycle checker. It's over 5 years old now, and this is not something worth including in every D program.

Ok, if no one is using it makes sense to remove it. We should, at least, leave a notice to remove this in the future and make it completely unrecognized.

File: test/cycles/Makefile (hunk 1 above)
Comment (ljmf00) [reply test/cycles/Makefile:17]: I was forgetting the fact that there's still a message being shown. Makes sense

**Comment by dkorpel** [comment]
> We should, at least, leave a notice to remove this in the future and make it completely unrecognized.

Done

**Pull request merged**